src/Microsoft.Health.Dicom.Core/Features/Telemetry/RetrieveMeter.cs

```diff
@@ -15,11 +16,24 @@ public sealed class RetrieveMeter : IDisposable
     public RetrieveMeter()
     {
         _meter = new Meter($"{OpenTelemetryLabels.BaseMeterName}.Retrieve", "1.0");
-        RetrieveInstanceCount = _meter.CreateCounter<long>(nameof(RetrieveInstanceCount), description: "Total number of instances retrieved");
+        RetrieveInstanceMetadataCount = _meter.CreateCounter<long>(nameof(RetrieveInstanceMetadataCount), description: "Count of metadata retrieved");
+        RetrieveInstanceCount = _meter.CreateCounter<long>(nameof(RetrieveInstanceCount), description: "Count of instances retrieved");
     }
 
+    public Counter<long> RetrieveInstanceMetadataCount { get; }
+
     public Counter<long> RetrieveInstanceCount { get; }
 
+    public static KeyValuePair<string, object>[] RetrieveInstanceCountTelemetryDimension(long contentLength, long maxFileSize, bool isTranscoding = false, bool hasFrameMetadata = false, bool isRendered = false) =>
+        new[]
+        {
+            new KeyValuePair<string, object>("ContentLength", contentLength),
+            new KeyValuePair<string, object>("MaxFileSize", maxFileSize),
```

Comment:
nit: It could be MaxFileSizeForTranscoding. But not sure why we want to log this info